src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/StatelessDataController.cs

```diff
@@ -72,66 +94,112 @@ public async Task<ActionResult> Post([FromQuery] string dataType)
                 return BadRequest($"Invalid dataType {dataType} provided. Please provide a valid dataType as query parameter.");
             }
 
-            object appModel = _altinnApp.CreateNewAppModel(classRef);
+            if (GetPartyHeader(HttpContext).Count > 1)
+            {
+                return BadRequest($"Invalid party. Only one allowed");
+            }
+
+            int? partyId = await GetPartyId(HttpContext);
+
+            EnforcementResult enforcementResult = await AuthorizeAction(org, app, partyId.Value, "read");
+
+            if (!enforcementResult.Authorized)
+            {
+                return Forbidden(enforcementResult);
+            }
 
-            int? partyId = HttpContext.User.GetPartyIdAsInt();
+            _altinnAppContext.SetContext(new AltinnAppContext() { PartyId = partyId.Value });
+
+            ModelDeserializer deserializer = new ModelDeserializer(_logger, _altinnApp.GetAppModelType(classRef));
+            object appModel = await deserializer.DeserializeAsync(Request.Body, Request.ContentType);
+            if (appModel == null)
```

Comment:
do you see any use cases where deserialization fails and we would like to pass the error message onto the client. e.g. something in the body is crashing the deserialization, rather than returning a new model? 
Not sure how often this would happen or what could trigger it, but I making a difference of appmodel being null because no data was provided and appmodel being null because an error occured could be useful

Reply:
Restructured with a new get method